modules/openapi-generator/src/main/resources/python/python-experimental/model_utils.mustache

```diff
@@ -1124,9 +1124,10 @@ def validate_get_composed_info(constant_args, model_args, self):
                 "class %s. Not all inputs were used. The unused input data "
                 "is %s" % (self.__class__.__name__, unused_args)
             )
-        for var_name, var_value in six.iteritems(unused_args):
-            for instance in additional_properties_model_instances:
-                setattr(instance, var_name, var_value)
+        else:
+            for var_name, var_value in six.iteritems(unused_args):
+                for instance in additional_properties_model_instances:
+                    setattr(instance, var_name, var_value)
```

Comment:
Can you explain why we need this?
I think that unused_args will be length 0 if we are dropping the extra kwargs before we invoke this method.

Reply:
No it won't be. This is the case when the unused_args are injected into the additionalproperties map.

Reply:
Right, but didn't we delete those args with:
self._configuration.discard_unknown_keys = True?
So they no longer exist here.

What should we do if:
- What if one of our composed models has additional properties in it + discard_unknown_keys=True?
- What if self model has additional properties defined in self and discard_unknown_keys=True?

Reply:
No, I removed that code. If additional properties exists, it will populate the undeclared properties into the additionalproperties Map. Then at that point there will be nothing to discard.

Reply:
So when discard_unknown_keys=True it still allows additionalproperties to process unknown keys?

Reply:
Yes, I think that should be the behavior.

Reply:
In line41 of modules/openapi-generator/src/main/resources/python/python-experimental/model_templates/method_init_composed.mustache we are already assigning additional properties.
How about deleting this else clause because we already handle assign these properties there?

Reply:
I've added documentation in the configuration file to clarify. There is a difference between "undeclared" versus unknown.

Reply:
Right but we already assign these properties in line41.
So right now this code is assigning properties in two places.
Can we only assign these properties in one place?
Can that be on line41 in `__init__`?

Reply:
> In line41 of modules/openapi-generator/src/main/resources/python/python-experimental/model_templates/method_init_composed.mustache we are already assigning additional properties.
> How about deleting this else clause because we already handle assign these properties there?

Sorry I am confused, at line 41 in method_init_composed.mustache there is no else clause

Reply:
I think I understand now. I've submitted another change.

Reply:
At that line we are looping over all kwargs.
Some of those are the unused args.
For those unused args we assign the key value properties to self.
Self puts those in its own data store, and the data stores of all composed models that contain that property.

Reply:
There is some strange delay in github message delivery. I think I have addressed your point.